### PR TITLE
[MM-166] Fix issue: decription not visible in tooltip and removed "see more" link

### LINKED ISSUE #734: Issue/PR descriptions not showing in link tooltips
CleanShot 2024-01-25 at 09 51 14
CleanShot 2024-01-25 at 09 51 24

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
#### Summary
- Fix issue: description not visible in tooltip.
- Removed the see more link in the tooltip.
- Set a max length for the tooltip description.

#### Screenshot
image

#### What to test
- Description and other contents are visible in the tooltip properly.
- "See more" link is removed
- When the description is very long it's getting trimmed.

###### Steps to test:
1. Connect your GitHub account
2. Hover over a PR/Issue link.

#### Ticket Link
Fixes #734

### CODE REVIEW

**Review by mickmister — COMMENTED** [review]
Great job @raghavaggarwal2308 :+1:  Looks good, just a few comments for discussion

File: webapp/src/components/link_tooltip/tooltip.css [hunk 1]
```diff
@@ -66,9 +66,7 @@
 }
 
 .github-tooltip .tooltip-info .markdown-text {
-    max-height: 150px;
     line-height: 1.25;
-    overflow: hidden;
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply webapp/src/components/link_tooltip/tooltip.css:71]: @mickmister This was being used to trim the description when it was too long. We modified the logic to set a max length to the description now.

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 2]
```diff
@@ -118,10 +120,7 @@ export const LinkTooltip = ({href, connected, show, theme}) => {
                                 <span>{'#' + data.number}</span>
                             </a>
                             <div className='markdown-text mt-1 mb-1'>
-                                <ReactMarkdown
-                                    source={data.body}
-                                    linkTarget='_blank'
```
[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply webapp/src/components/link_tooltip/link_tooltip.jsx:123]: @mickmister I misunderstood the use of this prop. Added it back

File: webapp/src/components/link_tooltip/link_tooltip.jsx [hunk 3]
```diff
@@ -87,6 +89,14 @@ export const LinkTooltip = ({href, connected, show, theme}) => {
         let date = new Date(data.created_at);
         date = date.toDateString();
 
+        let description = '';
+        if (data.body) {
+            description = data.body.substring(0, maxTicketDescriptionLength).trim();
+            if (description.length > maxTicketDescriptionLength) {
+                description += '...';
+            }
+        }
```
Comment (mickmister) [inline webapp/src/components/link_tooltip/link_tooltip.jsx:98]: I actually got this wrong in my previous suggestion. `description.length` will never be greater than `maxTicketDescriptionLength` in that block

```suggestion
        let description = '';
        if (data.body) {
            description = data.body.substring(0, maxTicketDescriptionLength).trim();
            if (data.body.length > maxTicketDescriptionLength) {
                description += '...';
            }
        }
```

File: webapp/src/components/link_tooltip/tooltip.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (mickmister) [reply webapp/src/components/link_tooltip/tooltip.css:71]: It's still possible that it overflows though right? Is there any risk in keeping `overflow: hidden`?

[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply webapp/src/components/link_tooltip/tooltip.css:71]: @mickmister I don't think it will overflow but there is no risk in keeping it so I have reverted the change

File: webapp/src/components/link_tooltip/tooltip.css [hunk 4]
```diff
@@ -66,13 +66,13 @@
...
     word-break: break-word;
     word-wrap: break-word;
     overflow-wrap: break-word;
     font-size: 12px;
+    max-height: 150px;
+    overflow: hidden;
```
Comment (hanzei) [inline webapp/src/components/link_tooltip/tooltip.css:75]: Why not completly revert the change? The diff is off now.

**Review by hanzei — APPROVED** [review]
LGTM

File: webapp/src/components/link_tooltip/tooltip.css (hunk 4 above)
Comment (raghavaggarwal2308) [reply webapp/src/components/link_tooltip/tooltip.css]: @hanzei Reverted

**Review by mickmister — APPROVED** [review]

File: webapp/src/components/link_tooltip/tooltip.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (mickmister) [reply webapp/src/components/link_tooltip/tooltip.css]: Was it the `overflow: hidden` that was making it so the description wasn't being shown at all?

[the remark this answers is not in the document]
Comment (raghavaggarwal2308) [reply webapp/src/components/link_tooltip/tooltip.css]: @mickmister No, the component `ReactMarkdown` was not used properly.

**Review by AayushChaudhary0001 — APPROVED** [review]
This PR has been tested for the following scenarios:-
-  Added description in the tooltip.
- Removed the "See more" link from the tooltip.
- Text gets trimmed if its length is too much.

The PR is working fine for the above conditions, LGTM. Approved.
